### PR TITLE
make some calls generic

### PR DESCRIPTION
**Pull request by apahl**
show, hide, enable, disable
(@chemist69 on IRC)

### CODE REVIEW

File: ui.nim [hunk 1]
```diff
@@ -467,6 +464,20 @@ proc newMenu*(name: string): Menu =
   result.impl = rawui.newMenu(name)
   result.children = @[]
 
+# -------------------- Generics ------------------------------------
+
+proc show*[W](w: W) =
```
Comment (Araq) [inline ui.nim]: Add a generic constraint like so '[W: Widget]'

**Comment by Araq** [comment]
Has conflicts with master, please rebase.

**Comment by apahl** [comment]
After your introduction of mainSteps in rawui.nim, my version of libui does not work anymore. I first have  to rebuild it but will probably not have time for that in the next days (I am currently using the pre-built alpha 3 binary). Closed the PR and wll try again later. Thanks again for your efforts with the libui wrapper!

**Comment by Araq** [comment]
I can upload binaries. Which OS? Windows?

**Comment by apahl** [comment]
Yes, Windows, thank you!
For Linux I could build it myself.
If possible, could you also show me how I can cross-compile libui on Linux for Windows?
I have the whole gcc cross-compilation toolchain set up, but I have so far only used it with Nim.
I have a Win7 VM, but no development toolchain on it (no admin rights on it;-). Therefore I do all my development on Linux and cross-compile to Windows. Nim and libui / ui are a perfect team for that!

**Pull request closed without merging**